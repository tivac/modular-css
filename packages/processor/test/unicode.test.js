"use strict";

const namer  = require("@modular-css/test-utils/namer.js");
const Processor = require("../processor.js");

describe("/processor.js", () => {
    describe("unicode", () => {
        let processor;
        
        beforeEach(() => {
            processor = new Processor({
                namer,
            });
        });

        it("should support unicode classes & ids", async () => {
            await processor.file(require.resolve("./specimens/unicode.css"));

            const { css } = await processor.output({
                to : "./packages/processor/test/output/unicode.css",
            });

            expect(css).toMatchSnapshot();
        });
    });
});
