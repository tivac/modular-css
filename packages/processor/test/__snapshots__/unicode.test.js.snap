// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/processor.js unicode should support unicode classes & ids 1`] = `
"/* packages/processor/test/specimens/unicode.css */
#mc_‚ô• { background: lime; }
#mc_¬© { background: lime; }
#mc_‚Äú‚Äò‚Äô‚Äù { background: lime; }
#mc_‚ò∫‚òÉ { background: lime; }
#mc_‚åò‚å• { background: lime; }
#mc_ùÑû‚ô™‚ô©‚ô´‚ô¨ { background: lime; }
#mc_üí© { background: lime; }
#\\\\? { background: lime; }
#\\\\@ { background: lime; }
#\\\\. { background: lime; }
#\\\\3A \\\\) { background: lime; }
#\\\\3A \\\\\`\\\\( { background: lime; }
#\\\\31 23 { background: lime; }
#\\\\31 a2b3c { background: lime; }
#\\\\<p\\\\> { background: lime; }
#\\\\<\\\\>\\\\<\\\\<\\\\<\\\\>\\\\>\\\\<\\\\> { background: lime; }
#\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\[\\\\>\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\>\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\>\\\\+\\\\+\\\\+\\\\>\\\\+\\\\<\\\\<\\\\<\\\\<\\\\-\\\\]\\\\>\\\\+\\\\+\\\\.\\\\>\\\\+\\\\.\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\.\\\\.\\\\+\\\\+\\\\+\\\\.\\\\>\\\\+\\\\+\\\\.\\\\<\\\\<\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\.\\\\>\\\\.\\\\+\\\\+\\\\+\\\\.\\\\-\\\\-\\\\-\\\\-\\\\-\\\\-\\\\.\\\\-\\\\-\\\\-\\\\-\\\\-\\\\-\\\\-\\\\-\\\\.\\\\>\\\\+\\\\.\\\\>\\\\. { background: lime; }
#\\\\# { background: lime; }
#\\\\#\\\\# { background: lime; }
#\\\\#\\\\.\\\\#\\\\.\\\\# { background: lime; }
#\\\\_ { background: lime; }
#\\\\{\\\\} { background: lime; }
#\\\\.fake\\\\-class { background: lime; }
#foo\\\\.bar { background: lime; }
#\\\\3A hover { background: lime; }
#\\\\3A hover\\\\3A focus\\\\3A active { background: lime; }
#\\\\[attr\\\\=value\\\\] { background: lime; }
#f\\\\/o\\\\/o { background: lime; }
#f\\\\\\\\o\\\\\\\\o { background: lime; }
#f\\\\*o\\\\*o { background: lime; }
#f\\\\!o\\\\!o { background: lime; }
#f\\\\'o\\\\'o { background: lime; }
#f\\\\~o\\\\~o { background: lime; }
#f\\\\+o\\\\+o { background: lime; }

.mc_\\\\2665 { background: lime; }
.mc_\\\\A9 { background: lime; }
.mc_\\\\201C\\\\2018\\\\2019\\\\201D { background: lime; }
.mc_\\\\263A\\\\2603 { background: lime; }
.mc_\\\\2318\\\\2325 { background: lime; }
.mc_\\\\1D11E\\\\266A\\\\2669\\\\266B\\\\266C { background: lime; }
.mc_\\\\1F4A9 { background: lime; }
.mc_\\\\? { background: lime; }
.mc_\\\\@ { background: lime; }
.mc_\\\\. { background: lime; }
.mc_\\\\:\\\\) { background: lime; }
.mc_\\\\:\\\\\`\\\\( { background: lime; }
.mc_123 { background: lime; }
.mc_1a2b3c { background: lime; }
.mc_\\\\<p\\\\> { background: lime; }
.mc_\\\\<\\\\>\\\\<\\\\<\\\\<\\\\>\\\\>\\\\<\\\\> { background: lime; }
.mc_\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\[\\\\>\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\>\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\>\\\\+\\\\+\\\\+\\\\>\\\\+\\\\<\\\\<\\\\<\\\\<-\\\\]\\\\>\\\\+\\\\+\\\\.\\\\>\\\\+\\\\.\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\.\\\\.\\\\+\\\\+\\\\+\\\\.\\\\>\\\\+\\\\+\\\\.\\\\<\\\\<\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\.\\\\>\\\\.\\\\+\\\\+\\\\+\\\\.------\\\\.--------\\\\.\\\\>\\\\+\\\\.\\\\>\\\\. { background: lime; }
.mc_\\\\# { background: lime; }
.mc_\\\\#\\\\# { background: lime; }
.mc_\\\\#\\\\.\\\\#\\\\.\\\\# { background: lime; }
.mc__ { background: lime; }
.mc_\\\\{\\\\} { background: lime; }
.mc_\\\\#fake-id { background: lime; }
.mc_foo\\\\.bar { background: lime; }
.mc_\\\\:hover { background: lime; }
.mc_\\\\:hover\\\\:focus\\\\:active { background: lime; }
.mc_\\\\[attr\\\\=value\\\\] { background: lime; }
.mc_f\\\\/o\\\\/o { background: lime; }
.mc_f\\\\\\\\o\\\\\\\\o { background: lime; }
.mc_f\\\\*o\\\\*o { background: lime; }
.mc_f\\\\!o\\\\!o { background: lime; }
.mc_f\\\\'o\\\\'o { background: lime; }
.mc_f\\\\~o\\\\~o { background: lime; }
.mc_f\\\\+o\\\\+o { background: lime; }
"
`;
